tests/cpgrid/inactiveCell_lgr_test.cpp

```diff
@@ -0,0 +1,583 @@
+                    BOOST_CHECK_EQUAL((std::find(std::get<1>((*data[0]).parent_to_children_cells_[child_to_parent[1]]).begin(),
+                                                 std::get<1>((*data[0]).parent_to_children_cells_[child_to_parent[1]]).end(),
+                                                 level_cellIdx[1]) ==
+                                       std::get<1>((*data[0]).parent_to_children_cells_[child_to_parent[1]]).end()) , false);
+                    // Check amount of children cells of the parent cell
+                    BOOST_CHECK_EQUAL(std::get<1>((*data[0]).parent_to_children_cells_[child_to_parent[1]]).size(),
+                                      cells_per_dim_vec[entity.level()-1][0]*
+                                      cells_per_dim_vec[entity.level()-1][1]*cells_per_dim_vec[entity.level()-1][2]);
+                    BOOST_CHECK( entity.father().isLeaf() == false);
+                    BOOST_CHECK( (entity.level() > 0) || (entity.level() < static_cast<int>(startIJK_vec.size()) +1));
+                    BOOST_CHECK( level_cellIdx[0] == entity.level());
+                }
+                else{
+                    BOOST_CHECK_THROW(entity.father(), std::logic_error);
+                    BOOST_CHECK_THROW(entity.geometryInFather(), std::logic_error);
+                    BOOST_CHECK_EQUAL(child_to_parent[0], -1);
+                    BOOST_CHECK_EQUAL(child_to_parent[1], -1);
+                    BOOST_CHECK( level_cellIdx[0] == 0);
+                    BOOST_CHECK( std::get<0>((*data[0]).parent_to_children_cells_[level_cellIdx[1]]) == -1);
+                    BOOST_CHECK( std::get<1>((*data[0]).parent_to_children_cells_[level_cellIdx[1]]).empty());
+                    BOOST_CHECK( entity.level() == 0);
+                    // Get index of the cell in level 0
+                    const auto& entityOldIdx =  (*data[startIJK_vec.size()+1]).leaf_to_level_cells_[entity.index()][1];
+                    BOOST_CHECK( entity.getOrigin().index() == entityOldIdx);
+                    BOOST_CHECK( entity.getOrigin().level() == 0);
+                    // Get IJK of the old index
+                    std::array<int,3> entityOldIJK;
+                    (*data[0]).getIJK(entityOldIdx, entityOldIJK); // ijk
+                }
+            }
+        } // end-level-for-loop
+
+        BOOST_CHECK( static_cast<int>(startIJK_vec.size()) == grid.maxLevel());
+        BOOST_CHECK( (*data[data.size()-1]).parent_to_children_cells_.empty());
+
+        for (long unsigned int l = 0; l < startIJK_vec.size() +1; ++l) // level 0,1,2,... , last patch
+        {
+            const auto& view = grid.levelGridView(l);
+            for (const auto& element: elements(view)){
+                BOOST_CHECK_EQUAL(element.level(), l);
+            }
+        }
+        //////////////////////////////
+    }
+    catch (const std::exception& e) {
+        std::cout << e.what() << "'\n";
```

Comment:
I still don't like this one.

Imagine that there is an exception in try block, then we only print an error here, but the test will seem fine to ctest. Either don't do try-catch (-> test fails if there is an exception) or `BOOST_CHECK(false)` here.

Reply:
I modified the test removing the try-catch, replacing it by BOOST_THROW in the last test-case "throw_allInactiveCells_in_lgr". I hope that this is kind of equivalent to what you suggested.